crates/bevy_ecs/src/schedule/runner/mod.rs

```diff
@@ -0,0 +1,190 @@
+mod multi_threaded;
+mod single_threaded;
+
+use crate::{
+    schedule::{BoxedRunCondition, RegistryId, SystemLabel, SystemRegistry},
+    system::BoxedSystem,
+    world::World,
+};
+
+use self::single_threaded::{SimpleRunner, SingleThreadedRunner};
+
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::Instrument;
+use bevy_utils::HashMap;
+
+use downcast_rs::{impl_downcast, Downcast};
+use fixedbitset::FixedBitSet;
+
+/// Types that can run [`System`] instances on the data stored in a [`World`].
+pub trait SystemRunner: Downcast + Send + Sync {
+    fn run(&mut self, world: &mut World);
+}
+
+impl_downcast!(SystemRunner);
+
+/// Internal resource used to signal the runner to apply pending commands to the [`World`].
+pub(super) struct RunnerApplyBuffers(pub bool);
+
+impl Default for RunnerApplyBuffers {
+    fn default() -> Self {
+        Self(false)
+    }
+}
+
+/// Applies pending command queues to the [`World`].
+pub fn apply_buffers(world: &mut World) {
```

Comment:
It's not clear that commands will always be the only type of buffers we have. If they are, we should rename this method (and the analogues) for clarity. If not, we should commands as an example in the docs, rather than a definition.

Reply:
I think my only option is to say the buffers include commands. To my knowledge, we don't have a way of *only* applying the command queues.